tests/basetest.rb

```diff
@@ -141,9 +141,16 @@ def config(*args)
     # Send the entire config as one string but be sure not to return until
     # we are safely back out of config mode, i.e. prompt is
     # 'switch#' not 'switch(config)#' or 'switch(config-if)#' etc.
-    @device.cmd('String' => "configure terminal\n" + args.join("\n") + "\nend",
-                # NX-OS has a space after '#', IOS XR does not
-                'Match'  => /^[^()]+[$%#>] *\z/n)
+    result = @device.cmd('String' => "configure terminal\n" + args.join("\n") +
+                         "\nend",
+                         # NX-OS has a space after '#', IOS XR does not
+                         'Match'  => /^[^()]+[$%#>] *\z/n)
+
+    if /invalid/i.match(result)
+      Cisco::Logger.warn("Config result:\n#{result}")
+    else
+      Cisco::Logger.debug("Config result:\n#{result}")
```

Comment:
We don't generally use 'info' level debugging in the main code because it makes Puppet too noisy by default - but since this is test-only code, would it make sense for this to use `info` level so as to provide an intermediate level of debugging for tests that's less verbose than the full `debug` level?